src/winning_check.c

```diff
@@ -0,0 +1,258 @@
+    }
+    if (count <= 5) {
+        if (count != 4) {
+            if (count<4){return count;}
+            return count+1;
+        } else {
+            if (value != 0) {
+                return 7;
+            } else return 2;
+        }
+    }
+    if (count == 6) {
+        if (value != 0) { return 7; }
+        if (how_many_set == 2) { return 6; }
+        else return 2;
+    }
+    return 0;
+}//
+void buble_sort_for_carts(card_t *carts) {
+    for (int i = 0; i < 6; ++i) {
+        for (int j = 0; j < 6; ++j) {
+            if (carts[j].value < carts[j + 1].value) {
+                card_t par = carts[j + 1];
+                carts[j + 1] = carts[j];
+                carts[j] = par;
+            }
+        }
+    }
+int is_in_sequence(card_t *carts, int *start_of_sequence) {
+    int count;
+    start_of_sequence[0] = carts[0].value;
+    for (int i = 0; i < 3; ++i) {
+        count = 1;
+        for (int j = 0; j < 6; ++j) {
+            if (carts[j].value == carts[j + 1].value + 1) {
+                count += 1;
+                if (count > 4) { return 1; }
+            }
+            if (carts[j].value != carts[j + 1].value + 1 && carts[j].value != carts[j + 1].value) {
+                start_of_sequence[0] = carts[j + 1].value;
+                count = 1;
+            }
+        }
+    }
+    if (count == 4 && carts[6].value == 2 && carts[0].value == 14) { return 1; }// check if ace take part in set
+    return 0;
+}
+
+void winner_check(player_t *players, int numbers_of_player, card_t table_carts[5], int *how_many_players_win, player_t *table_of_winners ) {
+    int player_score[numbers_of_player];//players score
+    for (int i = 0; i < numbers_of_player; ++i) {
+        player_score[i]=0;
+    }
+    int highest_cart_value[numbers_of_player];//highest cart in players hand// include table carts important in looking for winer without any set
+    int start_of_sequence[numbers_of_player];//cart that start sequence of "strit" // looking for winner
+    int power_of_pair_carts[numbers_of_player][4];//power of cart that make sets of pair or triple
+    int highest_color_cart[numbers_of_player];//highest card in color set/ if therw is not any color - 0
+    for (int i = 0; i < numbers_of_player; ++i) {
+        for (int j = 0; j < 4; ++j) {
+            power_of_pair_carts[i][j] = 0;
+        }
+    }
+    for (int i = 0; i < numbers_of_player; ++i) {
+        card_t player_cart[7];
+        for (int j = 0; j < 5; ++j) {
+            player_cart[j] = table_carts[j];
+        }
+        player_cart[5] = players[i].cards[0];
+        player_cart[6] = players[i].cards[1];
+        buble_sort_for_carts(player_cart);
+        highest_cart_value[i] = player_cart[0].value;
+        int is_sequence = is_in_sequence(player_cart, &start_of_sequence[i]);
+        int is_color = serching_for_color(player_cart, &highest_color_cart[i]);
+        int strength_of_pair = fstrength_of_pair(player_cart, &power_of_pair_carts[i][0]);
```

Comment:
you pass a pointer to a table: use `power_of_pair_carts[i]`, not `&power_of_pair_carts[i][0]`

Reply:
i want to pass a pointer to this function

Reply:
i want to pass a pointer to this function